src/opentelemetry.cr

```diff
@@ -28,6 +28,7 @@ module OpenTelemetry
   # ```
   def self.trace(name : String)
     trace = current_trace
+    trace.resource = CONFIG.resource
```

Comment:
Not only it won't access the ENV var on each trace anymore, it also won't instantiate a new `Proto::Resource::V1::Resource` per trace (AFAICT 😅) so it should have better performance than the initial approach.

I've been running this branch on a side project so I might have some data on this tomorrow (I'll share the difference if it's noticeable).

Reply:
> so there will be no heap allocations per trace resource

Good news then despite the copy of the `struct` (TIL thanks!) 😄